qcs/src/qpu/execution.rs

```diff
@@ -243,7 +260,12 @@ impl<'a> Execution<'a> {
                     })
             })
             .collect::<Result<Vec<f64>>>()?;
-        patch_values.insert(SUBSTITUTION_NAME, values);
+        // Convert back to the format that this library expects
+        let mut patch_values: Parameters = params
+            .into_iter()
+            .map(|(key, value)| (key.into(), value))
+            .collect();
+        patch_values.insert(SUBSTITUTION_NAME.into(), values);
```

Comment:
I don't think there is a test covering this, but I might have missed it. Might be good to add one, since based on the comments "this library expects" and "that quil-rs expects", we would benefit from those invariants being checked for regression over time.

Reply:
It's really a type conversion—this would not compile if it wasn't converting correctly. We have a `HashMap<Box<str>, Vec<f64>>` and `quil-rs` wants a `HashMap<&str, Vec<f64>>`. The comment is just to clarify the fact that this iteration looks kinda useless.

Is there a better comment I could leave to reduce confusion?

Reply:
I see -- that's good enough for me :) Thanks for clarifying.